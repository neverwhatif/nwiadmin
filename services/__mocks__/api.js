let resolveOrReject = true;

module.exports = {
    get: () => new Promise((resolve, reject) => {
        if (resolveOrReject) {
            resolve({ data: [{ id: 1, name: 'Lorem' }] });
        } else {
            reject();
        }
        resolveOrReject = !resolveOrReject;
    }),
    mockReset: () => {
        resolveOrReject = true;
    },
};
