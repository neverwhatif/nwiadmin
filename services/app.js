/* eslint-disable import/prefer-default-export */

import config from 'app/config';

export const setDocumentTitle = (title) => {
    document.title = title ? `${title} | ${config.app.name}` : config.app.name;
};
