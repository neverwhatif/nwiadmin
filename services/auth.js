import store from 'nwiadmin/utility/store';
import history from 'nwiadmin/utility/history';

import { stringifyRemote } from 'nwiadmin/utility';

const tokenName = 'token';
const authRouteName = 'login';

export const getToken = () => (store.getItem(tokenName) || '').replace(/\n/, '');
export const setToken = token => store.setItem(tokenName, token);
export const removeToken = () => store.removeItem(tokenName);

export const setTokenFromHeader = (headers = {}) => {
    const header = headers.authorization || headers.Authorization;

    if (header) {
        const token = header.replace(/^Bearer /, '');
        return setToken(token);
    }

    return null;
};

const getHeader = () => ({
    Authorization: `Bearer ${getToken()}`,
});

export const getAuthedHeaders = (headers = {}) => ({
    ...headers,
    ...getHeader(),
});

const handleUnauth = () => {
    removeToken();

    if (window.location.pathname === `/${authRouteName}`) {
        return;
    }

    const { pathname, search } = window.location;
    window.location = stringifyRemote([`/${authRouteName}`, { redirect: `${pathname}${search}` }]);
    //history.push(`/${authRouteName}`, { redirect: `${pathname}${search}` });
};

const checkForUnauthResponse = (error) => {
    if(!error.response) {
        console.log(error);
        return;
    }
    if (error.response.status === 401) {
        return true;
    }
    if (error.response.status === 400
        && error.response.data.error
        && error.response.data.error.match(/token/)
    ) {
        return true;
    }

    return false;
};

export const checkAuthResponse = (error, elseFn) => {
    // A bit hacky, probably should use some way of passing a 'isUnauth' parameter to the route or something
    if (window.location.pathname.match(/^\/reset/)) {
        return;
    }

    if (checkForUnauthResponse(error)) {
        handleUnauth();
        return;
    }

    if (elseFn && typeof elseFn === 'function') {
        elseFn();
    }
};

export const login = (token, redirect = '/') => {
    setToken(token);
    window.location = redirect;
};

export const logout = () => {
    removeToken();
    window.location = `/${authRouteName}`;
};
