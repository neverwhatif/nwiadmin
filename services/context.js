import React, { createContext, useContext } from 'react';

const AppContext = createContext({ me: null });

export const AppContextProvider = ({ children, value }) => (
    <AppContext.Provider value={value}>{children}</AppContext.Provider>
);

export const useAppContext = () => useContext(AppContext);
