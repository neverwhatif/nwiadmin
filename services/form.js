export const getFieldValue = (name, data) => (name ? name.split('.').reduce((acc, cur) => acc[cur], data) : null);

export const setUpdatedData = (name, value, data) => {
    const newData = { ...data };
    const parts = name.split(/\.(?=[^.]+$)/);

    if (parts.length === 1) {
        newData[parts[0]] = value;
        return newData;
    }

    getFieldValue(parts[0], newData)[parts[1]] = value;

    return newData;
};
