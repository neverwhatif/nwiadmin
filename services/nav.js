import routes from 'app/config/routes';

export const buildNestedPath = (basePath, path) => (
    `${basePath ? `/${basePath}` : ''}/${path || ''}`.replace(/(.+)\/$/, '$1').replace(/\/{2,}/g, '/')
);

export const checkPathIsActive = (path, locationPath, useFullLocationPath = true) => {
    if (useFullLocationPath) {
        return path === locationPath;
    }

    return path === `/${locationPath.split('/')[1]}`;
};

const parseNavItem = (item, index, locationPath, useFullLocationPath = true) => {
    const path = buildNestedPath(item.basePath, item.path);
    const isActive = checkPathIsActive(path, locationPath, useFullLocationPath);

    const children = item.children && item.children.length
        ? item.children.map((child, cIndex) => parseNavItem({ ...child, basePath: path }, cIndex, locationPath, true))
        : null;

    return {
        ...item,
        children,
        path,
        isActive,
        key: index,
    };
};

export const getPrimaryNav = locationPath => (
    routes.filter(item => Boolean(item.label)).map((item, index) => parseNavItem(item, index, locationPath, false))
);

export const parseTabs = (tabs, basePath, locationPath) => tabs.map((item, index) => parseNavItem({
    ...item,
    basePath,
}, index, locationPath));

