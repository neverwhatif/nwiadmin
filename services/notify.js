import PubSub from 'pubsub-js';

export const notify = (type, message) => {
    PubSub.publish('@notify/SEND', { type, message });
};

export const notifySuccess = message => notify('success', message);
export const notifyError = message => notify('error', message);
export const clearNotify = () => notify();
