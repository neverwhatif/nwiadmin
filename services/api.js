import axios from 'axios';
import qs from 'qs';

import config from 'app/config';
import { getAuthedHeaders, setTokenFromHeader } from './auth';

let responseTransformer = response => response;
let isRegistered = false;

export const registerAPI = (opts = {}) => {
    if (isRegistered) {
        console.error('API has already been registered');
        return;
    }

    if (opts.transformResponse) {
        responseTransformer = opts.transformResponse;
    }
    isRegistered = true;
};

const cancels = {};

const defaultOptions = {
    cancellable: false,
    headers: {},
};

const responseSuccess = (response) => {
    const transformed = responseTransformer(response);

    setTokenFromHeader(transformed.headers);
    return transformed.data;
};

const responseError = (error) => {
    if (axios.isCancel(error)) {
        return Promise.resolve(-1);
    }
    setTokenFromHeader(error.response ? error.response.headers : {});
    return Promise.reject(error);
};

const transformRequestToRaw = (requestObj, method) => {
    const fd = new FormData();
    const newRequestObj = { ...requestObj };

    Object.entries(requestObj.data).forEach((item) => {
        fd.append(item[0], item[1]);
    });

    if (method === 'patchraw') {
        fd.append('_method', 'patch');
    }

    newRequestObj.method = 'post';
    newRequestObj.data = fd;
    newRequestObj.headers['Content-Type'] = undefined;

    newRequestObj.transformRequest = $ => $;

    return newRequestObj;
};

export const makeUrl = (alias, params = {}) => `${config.paths.api}/${alias}${Object.keys(params).length
    ? `?${qs.stringify(params)}`
    : ''}`;

const createRequestConfig = (rawMethod, alias, params = {}, rawData = {}, rawOptions = {}) => {
    let method = rawMethod || 'post';

    const lowerMethod = method.toLowerCase();
    const data = { ...rawData };

    if (lowerMethod === 'delete' || lowerMethod === 'patch') {
        method = 'post';
        data._method = lowerMethod; // eslint-disable-line no-underscore-dangle
    }

    const options = {
        ...defaultOptions,
        ...rawOptions,
    };

    if (cancels[alias]) {
        cancels[alias].cancel();
    }

    const headers = getAuthedHeaders({
        ...options.headers,
        Accept: 'application/json',
    });

    let requestConfig = {
        headers,
        method,
        url: makeUrl(alias, params),
    };

    if (data) {
        requestConfig.data = data;
    }

    if (options.cancellable) {
        const { CancelToken } = axios;

        cancels[alias] = CancelToken.source();
        requestConfig.cancelToken = cancels[alias].token;
    }

    if (lowerMethod === 'postraw' || lowerMethod === 'patchraw') {
        requestConfig = transformRequestToRaw(requestConfig, lowerMethod);
    }

    return requestConfig;
};

export const request = (method, alias, params, data, options) => {
    const requestConfig = createRequestConfig(method, alias, params, data, options);
    return axios.request(requestConfig)
        .then(response => responseSuccess(response))
        .catch(error => responseError(error));
};

export const get = (alias, params, options) => request('get', alias, params, false, options);

export const post = (alias, params, data, options) => request('post', alias, params, data, options);
export const postRaw = (alias, params, data, options) => request('postraw', alias, params, data, options);

export const patch = (alias, params, data, options) => request('patch', alias, params, data, options);
export const patchRaw = (alias, params, data, options) => request('patchraw', alias, params, data, options);

export const apiDelete = (alias, params, data) => request('delete', alias, params, data);
