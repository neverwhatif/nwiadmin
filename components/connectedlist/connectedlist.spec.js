import React from 'react';
import { shallow } from 'enzyme';

import { mockReset } from 'nwiadmin/services/api';
import { ConnectedListComponent } from './index';

jest.mock('nwiadmin/components/emptylist', () => 'EmptyList');
jest.mock('nwiadmin/components/filterbar', () => 'FilterBar');
jest.mock('nwiadmin/components/list', () => 'List');
jest.mock('nwiadmin/components/loading', () => 'Loading');
jest.mock('nwiadmin/components/pagination', () => 'Pagination');
jest.mock('nwiadmin/components/errormessage', () => 'ErrorMessage');
jest.mock('nwiadmin/components/errorboundary', () => 'ErrorBoundary');

jest.mock('nwiadmin/services/api');
jest.mock('nwiadmin/utility');

const mockLocation = {
    pathname: '/lorem',
    search: '',
};

const mockFilters = () => (<div />);

const unLoadingWithDataState = {
    isDataLoading: false,
    data: [{ id: 1, name: 'Lorem' }],
    meta: [{ total: 1 }],
};

const unLoadingState = {
    isDataLoading: false,
};

const errorState = {
    hasDataError: true,
};

describe('ConnectedListComponent', () => {
    beforeEach(() => {
        mockReset();
    });

    const component = shallow(<ConnectedListComponent remote="lorem" location={mockLocation} />);
    const componentWithFilters =
        shallow(<ConnectedListComponent remote="lorem" location={mockLocation} filters={mockFilters} />);
    const componentWithPagination = shallow(<ConnectedListComponent remote="lorem" location={mockLocation} />);
    const componentLoading = shallow(<ConnectedListComponent remote="lorem" location={mockLocation} />);
    const componentEmpty = shallow(<ConnectedListComponent remote="lorem" location={mockLocation} />);
    const componentWithError = shallow(<ConnectedListComponent remote="lorem" location={mockLocation} />);

    componentWithFilters.setState(unLoadingWithDataState);
    componentWithPagination.setState(unLoadingWithDataState);
    componentEmpty.setState(unLoadingState);
    componentWithError.setState(errorState);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentWithFilters).toMatchSnapshot();
        expect(componentWithPagination).toMatchSnapshot();
        expect(componentLoading).toMatchSnapshot();
        expect(componentEmpty).toMatchSnapshot();
        expect(componentWithError).toMatchSnapshot();
    });
    it('should recieveprops', () => {
        jest.spyOn(ConnectedListComponent.prototype, 'getData');

        const newProps = { location: { pathname: '/lorem', search: 'lorem' } };

        component.setProps(newProps);
        expect(ConnectedListComponent.prototype.getData).toHaveBeenCalledTimes(1);

        // Attempt it again, and getData should not be called again (so will remain being called 1 time)
        component.setProps(newProps);
        expect(ConnectedListComponent.prototype.getData).toHaveBeenCalledTimes(1);
    });
    it('should render filters', () => {
        expect(componentWithFilters.find('FilterBar').length).toBe(1);
    });
    it('should render pagination', () => {
        expect(componentWithPagination.find('Pagination').length).toBe(1);
    });
    it('should render a loading icon', () => {
        expect(componentLoading.find('Loading').length).toBe(1);
    });
    it('should render an empty list', () => {
        expect(componentEmpty.find('EmptyList').length).toBe(1);
    });
    it('should render an errormessage', () => {
        expect(componentWithError.find('ErrorMessage').length).toBe(1);
    });
});
