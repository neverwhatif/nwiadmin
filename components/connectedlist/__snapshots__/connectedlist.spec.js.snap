// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedListComponent should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedListComponent
    cta={null}
    filterMap={Object {}}
    filters={null}
    hasSearch={true}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    renderList={[Function]}
    shouldOnlyUpdateWithFilters={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Loading />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "Loading",
      },
      false,
    ],
    "type": "ErrorBoundary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Loading />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "Loading",
        },
        false,
      ],
      "type": "ErrorBoundary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ConnectedListComponent should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedListComponent
    cta={null}
    filterMap={Object {}}
    filters={[Function]}
    hasSearch={true}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    renderList={[Function]}
    shouldOnlyUpdateWithFilters={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <React.Fragment>
          <FilterBar
            cta={null}
            filters={[Function]}
            hasSearch={true}
            isVisible={false}
            itemName={null}
            location={
              Object {
                "pathname": "/lorem",
                "search": "",
              }
            }
            setVisibility={[Function]}
          />
          <FilterSummary
            filterMap={Object {}}
            location={
              Object {
                "pathname": "/lorem",
                "search": "",
              }
            }
            resetFilters={[Function]}
          />
        </React.Fragment>,
        <List
          columns={Array []}
          data={
            Array [
              Object {
                "id": 1,
                "name": "Lorem",
              },
            ]
          }
          functions={
            Object {
              "reloadData": [Function],
              "updateData": [Function],
              "updateRow": [Function],
            }
          }
          isDisabled={false}
          shouldInitPreload={true}
          transformer={[Function]}
        />,
        <Pagination
          data={
            Array [
              Object {
                "total": 1,
              },
            ]
          }
          location={
            Object {
              "pathname": "/lorem",
              "search": "",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <FilterBar
              cta={null}
              filters={[Function]}
              hasSearch={true}
              isVisible={false}
              itemName={null}
              location={
                Object {
                  "pathname": "/lorem",
                  "search": "",
                }
              }
              setVisibility={[Function]}
            />,
            <FilterSummary
              filterMap={Object {}}
              location={
                Object {
                  "pathname": "/lorem",
                  "search": "",
                }
              }
              resetFilters={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cta": null,
              "filters": [Function],
              "hasSearch": true,
              "history": undefined,
              "isVisible": false,
              "itemName": null,
              "location": Object {
                "pathname": "/lorem",
                "search": "",
              },
              "setVisibility": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "FilterBar",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "filterMap": Object {},
              "location": Object {
                "pathname": "/lorem",
                "search": "",
              },
              "resetFilters": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "columns": Array [],
          "data": Array [
            Object {
              "id": 1,
              "name": "Lorem",
            },
          ],
          "functions": Object {
            "reloadData": [Function],
            "updateData": [Function],
            "updateRow": [Function],
          },
          "isDisabled": false,
          "shouldInitPreload": true,
          "transformer": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "List",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "data": Array [
            Object {
              "total": 1,
            },
          ],
          "location": Object {
            "pathname": "/lorem",
            "search": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "Pagination",
      },
    ],
    "type": "ErrorBoundary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <React.Fragment>
            <FilterBar
              cta={null}
              filters={[Function]}
              hasSearch={true}
              isVisible={false}
              itemName={null}
              location={
                Object {
                  "pathname": "/lorem",
                  "search": "",
                }
              }
              setVisibility={[Function]}
            />
            <FilterSummary
              filterMap={Object {}}
              location={
                Object {
                  "pathname": "/lorem",
                  "search": "",
                }
              }
              resetFilters={[Function]}
            />
          </React.Fragment>,
          <List
            columns={Array []}
            data={
              Array [
                Object {
                  "id": 1,
                  "name": "Lorem",
                },
              ]
            }
            functions={
              Object {
                "reloadData": [Function],
                "updateData": [Function],
                "updateRow": [Function],
              }
            }
            isDisabled={false}
            shouldInitPreload={true}
            transformer={[Function]}
          />,
          <Pagination
            data={
              Array [
                Object {
                  "total": 1,
                },
              ]
            }
            location={
              Object {
                "pathname": "/lorem",
                "search": "",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <FilterBar
                cta={null}
                filters={[Function]}
                hasSearch={true}
                isVisible={false}
                itemName={null}
                location={
                  Object {
                    "pathname": "/lorem",
                    "search": "",
                  }
                }
                setVisibility={[Function]}
              />,
              <FilterSummary
                filterMap={Object {}}
                location={
                  Object {
                    "pathname": "/lorem",
                    "search": "",
                  }
                }
                resetFilters={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cta": null,
                "filters": [Function],
                "hasSearch": true,
                "history": undefined,
                "isVisible": false,
                "itemName": null,
                "location": Object {
                  "pathname": "/lorem",
                  "search": "",
                },
                "setVisibility": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "FilterBar",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "filterMap": Object {},
                "location": Object {
                  "pathname": "/lorem",
                  "search": "",
                },
                "resetFilters": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "columns": Array [],
            "data": Array [
              Object {
                "id": 1,
                "name": "Lorem",
              },
            ],
            "functions": Object {
              "reloadData": [Function],
              "updateData": [Function],
              "updateRow": [Function],
            },
            "isDisabled": false,
            "shouldInitPreload": true,
            "transformer": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "List",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data": Array [
              Object {
                "total": 1,
              },
            ],
            "location": Object {
              "pathname": "/lorem",
              "search": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "Pagination",
        },
      ],
      "type": "ErrorBoundary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ConnectedListComponent should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedListComponent
    cta={null}
    filterMap={Object {}}
    filters={null}
    hasSearch={true}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    renderList={[Function]}
    shouldOnlyUpdateWithFilters={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <List
          columns={Array []}
          data={
            Array [
              Object {
                "id": 1,
                "name": "Lorem",
              },
            ]
          }
          functions={
            Object {
              "reloadData": [Function],
              "updateData": [Function],
              "updateRow": [Function],
            }
          }
          isDisabled={false}
          shouldInitPreload={true}
          transformer={[Function]}
        />,
        <Pagination
          data={
            Array [
              Object {
                "total": 1,
              },
            ]
          }
          location={
            Object {
              "pathname": "/lorem",
              "search": "",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "columns": Array [],
          "data": Array [
            Object {
              "id": 1,
              "name": "Lorem",
            },
          ],
          "functions": Object {
            "reloadData": [Function],
            "updateData": [Function],
            "updateRow": [Function],
          },
          "isDisabled": false,
          "shouldInitPreload": true,
          "transformer": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "List",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "data": Array [
            Object {
              "total": 1,
            },
          ],
          "location": Object {
            "pathname": "/lorem",
            "search": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "Pagination",
      },
    ],
    "type": "ErrorBoundary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <List
            columns={Array []}
            data={
              Array [
                Object {
                  "id": 1,
                  "name": "Lorem",
                },
              ]
            }
            functions={
              Object {
                "reloadData": [Function],
                "updateData": [Function],
                "updateRow": [Function],
              }
            }
            isDisabled={false}
            shouldInitPreload={true}
            transformer={[Function]}
          />,
          <Pagination
            data={
              Array [
                Object {
                  "total": 1,
                },
              ]
            }
            location={
              Object {
                "pathname": "/lorem",
                "search": "",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "columns": Array [],
            "data": Array [
              Object {
                "id": 1,
                "name": "Lorem",
              },
            ],
            "functions": Object {
              "reloadData": [Function],
              "updateData": [Function],
              "updateRow": [Function],
            },
            "isDisabled": false,
            "shouldInitPreload": true,
            "transformer": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "List",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data": Array [
              Object {
                "total": 1,
              },
            ],
            "location": Object {
              "pathname": "/lorem",
              "search": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "Pagination",
        },
      ],
      "type": "ErrorBoundary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ConnectedListComponent should render 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedListComponent
    cta={null}
    filterMap={Object {}}
    filters={null}
    hasSearch={true}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    renderList={[Function]}
    shouldOnlyUpdateWithFilters={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Loading />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "Loading",
      },
      false,
    ],
    "type": "ErrorBoundary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Loading />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "Loading",
        },
        false,
      ],
      "type": "ErrorBoundary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ConnectedListComponent should render 5`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedListComponent
    cta={null}
    filterMap={Object {}}
    filters={null}
    hasSearch={true}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    renderList={[Function]}
    shouldOnlyUpdateWithFilters={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <EmptyList />,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "description": undefined,
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "EmptyList",
      },
      null,
    ],
    "type": "ErrorBoundary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <EmptyList />,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "description": undefined,
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "EmptyList",
        },
        null,
      ],
      "type": "ErrorBoundary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ConnectedListComponent should render 6`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectedListComponent
    cta={null}
    filterMap={Object {}}
    filters={null}
    hasSearch={true}
    itemName={null}
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
    remote="lorem"
    renderList={[Function]}
    shouldOnlyUpdateWithFilters={false}
    transformer={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <Loading />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "Loading",
      },
      false,
    ],
    "type": "ErrorBoundary",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <Loading />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "Loading",
        },
        false,
      ],
      "type": "ErrorBoundary",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
